### ISSUE TITLE
False positive of DMI_RANDOM_USED_ONLY_ONCE

### ISSUE THREAD

**Issue by victornoel** [spotbugs/spotbugs#1806]
Repro with 4.5.0

```java
import java.util.Random;

public final class Repro {

    private final Random random;

    public Repro(long seed) {
        this.random = new Random(seed);
        var maxWidth = 10;
        for (int i = 0; i < 10; i++) {
            int max = random.nextInt(maxWidth) + 1;
            for (int j = 0; j < max; j++) {
                double solution = random.nextDouble() * 1600 - 800;
                System.out.println(solution + " / " + random.nextDouble());
            }
        }
    }
}
```

This raises DMI_RANDOM_USED_ONLY_ONCE on line 13 and 15.

**Comment by mazurkin** [comment]
Same for me with 4.7.3, the reason to use just one call with a random with seed is to provide stable randomization (same randomization for the same seed)
